test/subscriber.ts

```diff
@@ -708,20 +714,56 @@ describe('Subscriber', () => {
       };
 
       // Receive message and assert that it was exported
-      const stream: FakeMessageStream = stubs.get('messageStream');
-      stream.emit('data', pullResponse);
-      assert.ok(exporter.getFinishedSpans());
+      const msgStream = stubs.get('messageStream');
+      msgStream.emit('data', pullResponse);
+
+      const spans = exporter.getFinishedSpans();
+      assert.strictEqual(spans.length, 1);
+      const firstSpan = spans.concat().shift();
+      assert.ok(firstSpan);
+      assert.strictEqual(firstSpan.parentSpanId, parentSpanContext.spanId);
+      assert.strictEqual(
+        firstSpan.name,
+        `${subscriber.name} receive`,
+        'name of span should match'
+      );
+      assert.strictEqual(
+        firstSpan.kind,
+        SpanKind.CONSUMER,
+        'span kind should be CONSUMER'
+      );
+      assert.strictEqual(
+        firstSpan.attributes[GeneralAttribute.NET_PEER_NAME],
+        subscription.projectId,
+        'span net peer name should match'
+      );
+      assert.strictEqual(
+        firstSpan.attributes[MessagingAttribute.MESSAGING_OPERATION],
+        'process',
+        'span messaging operation should match'
+      );
+      assert.strictEqual(
+        firstSpan.attributes[MessagingAttribute.MESSAGING_SYSTEM],
+        'pubsub'
+      );
+      assert.strictEqual(
+        firstSpan.attributes[MessagingAttribute.MESSAGING_MESSAGE_ID],
+        messageWithSpanContext.message.messageId,
+        'span messaging id should match'
+      );
+      assert.strictEqual(
+        firstSpan.attributes[MessagingAttribute.MESSAGING_DESTINATION],
+        subscriber.name,
+        'span messaging destination should match'
+      );
+      assert.strictEqual(
+        firstSpan.attributes[MessagingAttribute.MESSAGING_DESTINATION_KIND],
+        'topic'
+      );
```

Comment:
All this detail and matching testing is much appreciated!

Reply:
Thank you! If you know how to get the project id in the `Publisher` that would be awesome to add.

Reply:
It could probably be piped through, but I guess I wouldn't worry about it too much? Getting everything working again seems like a good short term goal.

Reply:
Yeah, I will remove it for now :)

Reply:
Removed the NET_PEER_NAME attribute for now rest is still there :)